versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/tiered/builder/Ref.java

```diff
@@ -123,106 +111,164 @@
   Ref name(String name);
 
   /**
-   * Set the commit of the reference.
-   * <p>Must be called exactly once for {@link RefType#TAG}, never for {@link RefType#BRANCH}.</p>
-   *
-   * @param commit The commit of the reference.
-   * @return This consumer.
+   * Specifies that this reference is a tag, not a branch.
+   * <p>Either this function or {@link #branch()} must be called.</p>
+   * @return consumer instance for tags.
    */
-  Ref commit(Id commit);
+  Tag tag();
 
   /**
-   * Set the metadata of the reference.
-   * <p>Must be called exactly once for {@link RefType#BRANCH}, never for {@link RefType#TAG}.</p>
-   *
-   * @param metadata The metadata of the reference.
-   * @return This consumer.
+   * Specifies that this reference is a branch, not a tag.
+   * <p>Either this function or {@link #tag()} must be called.</p>
+   * @return consumer instance for branches.
    */
-  Ref metadata(Id metadata);
+  Branch branch();
 
-  /**
-   * Set the children of the reference.
-   * <p>Must be called exactly once for {@link RefType#BRANCH}, never for {@link RefType#TAG}.</p>
-   *
-   * @param children The children of the reference.
-   * @return This consumer.
-   */
-  Ref children(Stream<Id> children);
+  interface Tag {
+    /**
+     * Set the commit of the reference.
+     * <p>Must be called exactly once.</p>
+     *
+     * @param commit The commit of the reference.
+     * @return This consumer.
+     */
+    Tag commit(Id commit);
 
-  /**
-   * Set the commits of the reference.
-   * <p>Must be called exactly once for {@link RefType#BRANCH}, never for {@link RefType#TAG}.</p>
-   * <p>Implementations must immediately execute the callbacks in the {@link BranchCommitConsumer}
-   * to construct the commit-log of the branch.</p>
-   *
-   * @param commits The branch's commit-log receiver.
-   * @return This consumer.
-   */
-  Ref commits(Consumer<BranchCommitConsumer> commits);
+    /**
+     * Convenience method to get back to the {@link Ref}.
+     * @return owning {@link Ref}
+     */
+    Ref toRef();
```

Comment:
I think we should maybe call this backToRef, returnToRef, parentRef or similar. "To" functions usually are verb/conversion functions. E.g. toBinary() or toJson(), not returns to parent pointers.